Presentation.Web/Controllers/API/ItSystemUsageMigrationController.cs

```diff
@@ -109,18 +109,49 @@ private static ItSystemUsageMigrationDTO Map(ItSystemUsageMigration input)
                 FromSystem = input.FromItSystem.MapToNamedEntityDTO(),
                 ToSystem = input.ToItSystem.MapToNamedEntityDTO(),
                 AffectedItProjects = input.AffectedProjects.Select(DTOMappingExtensions.MapToNamedEntityDTO).ToList(),
-                AffectedContracts = input.AffectedContracts.Select(Map).ToList()
+                AffectedContracts = input.AffectedContracts.Select(DTOMappingExtensions.MapToNamedEntityDTO).ToList(),
+                AffectedRelations = input.AffectedRelations.Select(Map).ToList()
             };
         }
+        
 
-        private static ItSystemUsageContractMigrationDTO Map(ItContractMigration input)
+        private static RelationMigrationDTO Map(SystemRelationMigration input)
         {
-            return new ItSystemUsageContractMigrationDTO
+            var relation = input.Relation;
+            if (relation.RelationInterfaceId == null)
             {
-                Contract = input.Contract.MapToNamedEntityDTO(),
-                SystemAssociatedInContract = input.SystemAssociatedInContract,
-                AffectedInterfaceUsages = input.AffectedInterfaceUsages.Select(Map).ToList(),
-                InterfaceExhibitUsagesToBeDeleted = input.ExhibitUsagesToBeDeleted.Select(Map).ToList()
+                if (relation.AssociatedContractId == null)
+                {
+                    return new RelationMigrationDTO
+                    {
+                        sourceSystem = relation.FromSystemUsage.MapToNamedEntityDTO(),
```

Comment:
@JacobMalling Brug null propagation i stedet for alle de der mærkelige cases Du behøves kun 1 mapping. Det her er rod.

Hvis du vil skrive en metode der "måske" mapper hvis ikke den er null, skal du bruge null propagation.